core/src/main/scala/cromwell/core/PathFactory.scala

```diff
@@ -0,0 +1,110 @@
+package cromwell.core
+
+import java.io.Writer
+import java.nio.file.{FileSystem, Path}
+
+import scala.collection.immutable.Queue
+import scala.util.{Success, Try}
+
+object PathFactory {
+  private def findFileSystem(rawString: String, fss: List[FileSystem], mapping: PartialFunction[FileSystem, Try[Path]]) = {
```

Comment:
The reasoning was that It's not necessarily intuitive to a caller of this method why they should provide a `Stream` of FileSystems, it seemed more an implementation detail of the method itself. But I agree it seems better to pass in a `Stream` if that's what we need so I don't really mind.